src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -154,10 +155,17 @@ public ModelingSubmission save(ModelingSubmission modelingSubmission, ModelingEx
 
         Optional<Participation> optionalParticipation = participationService.findOneByExerciseIdAndStudentLoginAnyState(modelingExercise.getId(), username);
         if (!optionalParticipation.isPresent()) {
-            throw new EntityNotFoundException("No participation found for " + username + " in exercise " + modelingExercise.getId());
+            throw new EntityNotFoundException("No participation found for " + username + " in exercise with id " + modelingExercise.getId());
         }
         Participation participation = optionalParticipation.get();
 
+        // TODO: enable retry mechanism again
```

Comment:
```suggestion
        // For now, we do not allow students to retry their modeling exercise after they have received feedback, because this could lead to unfair situations. Some students might get the manual feedback early and can then retry the exercise within the deadline and have a second chance, others might get the manual feedback late and would not have a chance to try it out again.
        // TODO: think about how we can enable retry again in the future in a fair way
```